tern/analyze/docker/dockerfile.py

```diff
@@ -118,6 +125,27 @@ def parse_from_image(dfobj):
     return image_list
 
 
+def expand_from_images(dfobj):
+    '''Replace all parent_image values with their full tag@sha:digest value.
+    Update the structure dictionary with the same information.
+    dfobj: the Dockerfile object created using get_dockerfile_obj'''
+    # update parent_images
+    images = parse_from_image(dfobj)
+    for i, img in enumerate(images):
+        if general.check_image_string(img['name'] + tag_separator +
+                                      img['tag']):
```

Comment:
IIRC the dictionary returned by `parse_from_image` will have a non-empty `digest_type` and `digest` if that was the image in the Dockerfile, in which case, you don't have to get the digest. So you can just check if the dictionary has an `image` and a `tag`. If there is no `tag` then you can substitute that value with `latest` and run your `get_image` function.